src/main/scala/uk/gov/ons/bi/dataload/exports/HmrcBiCsvExtractor.scala

```diff
@@ -82,24 +39,77 @@ object HmrcBiCsvExtractor {
     log.info(s"BI index file contains ${biData.count} records.")
 
     // Extract the different sets of data we want, and write to output files
-    val legalEntities = getLegalEntities(biData)
+
+    val hmrcOutput = getHMRCOutput(biData, hmrcFile)
+    log.info(s"Writing ${hmrcOutput.count} Legal Entities to $hmrcFile")
+
+    val legalEntities = getLegalEntities(biData, legalFile)
     log.info(s"Writing ${legalEntities.count} Legal Entities to $legalFile")
-    BiCsvWriter.writeCsvOutput(legalEntities, legalFile)
 
-    val vat = getVatExploded(biData)
+    val vat = getVatExploded(biData, vatFile)
     log.info(s"Writing ${vat.count} VAT entries to $vatFile")
-    BiCsvWriter.writeCsvOutput(vat, vatFile)
 
-    val paye =  getPayeExploded(biData)
+    val paye =  getPayeExploded(biData, payeFile)
     log.info(s"Writing ${paye.count} PAYE entries to $payeFile")
-    BiCsvWriter.writeCsvOutput(paye, payeFile)
-
-    val hmrcOut = getHMRCOutput(biData)
-    log.info(s"Writing ${hmrcOut.count} hmrcOut entries to $hmrcFile")
-    BiCsvWriter.writeCsvOutput(hmrcOut, hmrcFile)
 
     // Clear cache
     biData.unpersist(false)
   }
 
+  def stringifyArr(stringArr: Column) = concat(lit("["), concat_ws(",", stringArr), lit("]"))
+
+  def getHMRCOutput(df: DataFrame, outputPath: String): DataFrame = {
+
+    val arrDF = df
+      .withColumn(DataFrameColumn.VatStringArr, df(DataFrameColumn.VatID).cast(ArrayType(StringType)))
+      .withColumn(DataFrameColumn.PayeStringArr, df(DataFrameColumn.PayeID).cast(ArrayType(StringType)))
+
+    val dropDF = arrDF
+      .withColumn(DataFrameColumn.VatString, stringifyArr(arrDF(DataFrameColumn.VatStringArr)))
+      .withColumn(DataFrameColumn.PayeString, stringifyArr(arrDF(DataFrameColumn.PayeStringArr)))
+      .drop(DataFrameColumn.VatID,DataFrameColumn.PayeID,DataFrameColumn.VatStringArr, DataFrameColumn.PayeStringArr)
+
+    val hmrcOutput = dropDF
+      .select("id","BusinessName","TradingStyle","PostCode",
+        "Address1", "Address2","Address3","Address4","Address5",
+        "IndustryCode","LegalStatus","TradingStatus",
+        "Turnover","EmploymentBands","CompanyNo","VatRef","PayeRef")
+
+    BiCsvWriter.writeCsvOutput(hmrcOutput, outputPath)
+    hmrcOutput
+  }
+
+  def getLegalEntities(df: DataFrame, outputPath: String): DataFrame = {
+
+    val legalEntities = df.select("id","BusinessName","TradingStyle","PostCode",
+      "Address1", "Address2","Address3","Address4", "Address5",
+      "IndustryCode","LegalStatus","TradingStatus",
+      "Turnover","EmploymentBands","CompanyNo")
+
+    BiCsvWriter.writeCsvOutput(legalEntities, outputPath)
```

Comment:
If you pull the write method out of this function you'll make the function much easier to test. It also means the method has a single responsibility which is cleaner.